### DISCUSSION TITLE
Enable Multi Hierachy/View

### DISCUSSION THREAD

**Discussion by thomasstuempfig** [mar10/wunderbaum#3] [General]
Hello together,
i already loved Fancytree. 
Though I would like to be able to use an arbitrary Array property representing the parent-child relationship instead of the fixed "children" property. This would avoid to go through the datastructure returned by a service.
It would be even better to consider a configurable list of properties beeing considered as children here ["sons","daughters"].

regards
Thomas

```
{
  type:"Male",
  name:"Thomas",
  sons:[
    {
      type:"Man",
      name:"John"
    }
  ],
  daughters:[
    {
      type:"Woman",
      name:"Mary"
    },
    
    {
      type:"Woman",
      name:"Daisy"
    }
  ]
}
```

**Comment by mar10** [comment]
I am not sure if I understand the request.

1. is it about renaming the `children: [...]` JSON property, with an arbitrary, configurable name?
2. Is it about a new JSON format with parent-child relationships as 'parent-referencing list'?
3. Is it about allowing multiple  arbitrary, configurable names that replace `children`?

Ad 1.
If the server sends a custom, incompatible format, I think it is acceptable to either fix it there (recommend, but not always possible).
Otherwise, the response should be converted in the `receive` event.
Seems straight forward to me, especially because renaming the 'children' property is one of many things that can be different from the standard format.

Ad 2.
This is planned. I am planning to add an alternative format, that comes as flat list of nodes that have a `parent: ID` property

Ad 3.
I don't understand this. How would you expect this data to be represented internally, and how should It be displayed.
What is the use-case / benefit?
